lib/preflight/getCombinedConnectionStats.ts

```diff
@@ -0,0 +1,141 @@
+
+  // packets, packetsLost:
+  //              subscriber, audio, inbound-rtp,
+  //              subscriber, video, inbound-rtp
+  const packets = getStatValues(subscriberStats, 'packetsReceived', ['audio', 'video'], ['inbound-rtp']).reduce((a, b) => a + b, 0);
+  const packetsLost = getStatValues(subscriberStats, 'packetsLost', ['audio', 'video'], ['inbound-rtp']).reduce((a, b) => a + b, 0);
+
+  // roundTripTime: publisher, audio, remote-inbound-rtp
+  //                publisher, video, remote-inbound-rtp
+  const trackRoundTripTime = getStatValues(publisherStats, 'roundTripTime', ['audio', 'video'], ['remote-inbound-rtp']).reduce((a, b) => Math.max(a, b), 0);
+
+  // currentRoundTripTime. subscriber, 'candidate-pair'
+  const currentRoundTripTime = getStatValues(subscriberStats, 'currentRoundTripTime', [], ['candidate-pair']).reduce((a, b) => Math.max(a, b), 0);
+  const roundTripTime = (currentRoundTripTime || trackRoundTripTime) * 1000;
+
+  const bytesSent = getStatValues(publisherStats, 'bytesSent', [], ['candidate-pair']).reduce((a, b) => a + b, 0);
+  const bytesReceived = getStatValues(subscriberStats, 'bytesReceived', [], ['candidate-pair']).reduce((a, b) => a + b, 0);
+
+  const selectedIceCandidatePairStats = extractSelectedActiveCandidatePair(subscriberStats);
+
+  const iceCandidateStats: RTCIceCandidateStats[] = [];
+  subscriberStats.forEach(stat => {
+    if (stat.type === 'local-candidate' || stat.type === 'remote-candidate') {
+      iceCandidateStats.push(makeStandardCandidateStats(stat));
+    }
+  });
+  return { timestamp, jitter, packets, packetsLost, roundTripTime, bytesSent, bytesReceived, selectedIceCandidatePairStats, iceCandidateStats };
+}
+
+
+function makeStandardCandidateStats(input: any) : RTCIceCandidateStats {
+  const standardizedCandidateStatsKeys = [
+    { key: 'transportId', type: 'string' },
+    { key: 'candidateType', type: 'string' },
+    { key: 'port', altKeys: ['portNumber'], type: 'number' },
+    { key: 'address', altKeys: ['ip', 'ipAddress'], type: 'string' },
+    { key: 'priority', type: 'number' },
+    { key: 'protocol', altKeys: ['transport'], type: 'string' },
+    { key: 'url', type: 'string' },
+    { key: 'deleted', type: 'boolean' },
+    { key: 'relayProtocol', type: 'string' },
+  ];
+
+  return standardizedCandidateStatsKeys.reduce(function(report: any, keyInfo) {
+    let keysToLookFor = [keyInfo.key];
+    if (keyInfo.altKeys) {
+      keysToLookFor = keysToLookFor.concat(keyInfo.altKeys);
+    }
+    var key = keysToLookFor.find(key => key in input);
+    if (key && typeof input[key] === keyInfo.type) {
+      report[keyInfo.key] = input[key];
+    }
+
+    return report;
+  }, {});
+}
+
+function extractSelectedActiveCandidatePair(stats: RTCStatsReport) : SelectedIceCandidatePairStats | null {
+  let activeCandidatePairStatsFound: RTCIceCandidatePairStats|null = null;
+  stats.forEach(statPair => {
+    if (statPair.type === 'candidate-pair' && statPair.nominated) {
```

Comment:
I think `nominated` is not supported on some browsers. Here's a cross browser solution I did for voice. Let me know if this works or if this is applicable to video. See whole method [here](https://github.com/twilio/twilio-client.js/blob/master/lib/twilio/rtc/stats.js#L56).

```js
const selectedCandidatePairReport = candidatePairs.find(pair =>
      // Firefox
      pair.selected ||
      // Spec-compliant way
      (transport && pair.id === transport.selectedCandidatePairId)
    );
```

Reply:
Thanks @charliesantos - I like your solution better, I will update.